packages/blocks/src/api/raw-handling/ms-newline-normaliser.js

```diff
@@ -0,0 +1,11 @@
+export default function( node ) {
+	if ( node.nodeName !== 'P' ) {
+		return;
+	}
+
+	if ( ! [ ...node.classList ].find( ( className ) => className.match( /^Mso[A-Z].*/ ) ) ) {
+		return;
+	}
+
+	node.innerHTML = node.innerHTML.replace( /(?<!<br>)\n/g, ' ' );
```

Comment:
Are we trying to strip line breaks (`\n`)? Why should it not be preceded by a line break element?

Reply:
Well, yes! 🙂 This is exactly what this is about.

Please refer to the [initial comment of the related issue](https://github.com/WordPress/gutenberg/issues/13497#issue-402976972).

This behavior, as well as the fix/handling for it, is specific to **pasting** content from MS Word. As you can see in the mentioned issue, MS Word copies content formatted with a maximum line-length of 80 characters. So there are newlines injected by MS Word itself. Any actual user-defined linebreak is actually a `<br>`, followed by a newline. And we do **not** want to replace that.

Does that make sense?